util/contentutil/multiprovider.go

```diff
@@ -85,6 +85,26 @@ func (mp *MultiProvider) ReaderAt(ctx context.Context, desc ocispecs.Descriptor)
 	return mp.base.ReaderAt(ctx, desc)
 }
 
+// Info returns a content.Info
+func (mp *MultiProvider) Info(ctx context.Context, dgst digest.Digest) (content.Info, error) {
+	type info interface {
+		Info(context.Context, digest.Digest) (content.Info, error)
+	}
+
+	mp.mu.RLock()
+	if p, ok := mp.sub[dgst]; ok {
+		mp.mu.RUnlock()
+		if ci, ok := p.(info); ok {
```

Comment:
`mp.mu.RUnlock()` is called twice when this is not `ok`

Reply:
Good catch, thanks!